# Project-Network-Analysis-in-Metagenomic

This repository contains a full Python implementation of a microbial network-inference workflow inspired by the analytical pipeline presented in the E-WADES study, which investigates ecological relationships between metagenome-assembled genomes (MAGs) across multiple geographic sites, offering a critical analysis of the mathematical and statistical methods used in the construction  of microbial co-occurrence networks based on metagenomic data. 

Unlike the original study (which starts from raw sequencing reads) this implementation begins with the processed abundance matrices provided by the authors of the referenced paper as supplementary material, and corresponding to those used in Figure 1b. These tables allow us to to concentrate on the network-oriented components of the analysis (namely correlation inference, adjacency matrix construction, signed network generation, and topological evaluation), avoiding the conversion of raw data into relative abundances or the zero-replacement step using pseudo-counts (typically required prior to log-ratio transformations).

The code starts by loading the abundance matrix in order to apply Centered Log-Ratio (CLR) transformation, which represents the actual starting point of the implemented workflow. Then, the inference step is performed by applying Pearson Correlation corrected with FDR on CLR-data: we obtain a weighted, signed and sparse Adjacency Matrix from which we build our network. Then, we perform topological analysis on our graphs, focusing on community detection and hub-taxa identification, by implementing Signed Modularity and some Centrality Measures (Closeness, Betweenness and  Spectral). To conclude the analysis, we apply Jaccard Index to compare network similarity across geographic locations.

The overarching aim is to reconstruct microbial association networks at multiple sampling sites and to use graph-theoretical tools to evaluate which taxa co-occur or compete, how microbial communities organize into modules or clusters, which MAGs act as hubs, connectors, or stability-determining taxa and how similar or different the ecological networks are across geographic sites.â€¨

The final results confirms several biological and topological properties described in the E-WADES study, such as the sparsity of the networks (typical for metagenomic datasets), the high modularity (indicating strong ecological compartmentalization) and the clear separation of microbial assemblages across sites.
